### DISCUSSION TITLE
Clarify Bullet versioning

### DISCUSSION THREAD

**Discussion by akien-mga** [bulletphysics/bullet3#3823] [General]
As a downstream user (Godot) and Linux package maintainer (Mageia), I'm getting increasingly confused by Bullet's versioning, and the fact that every single tagged release seems to miss bumping the version number in the files where it's defined.

See e.g. https://github.com/bulletphysics/bullet3/pull/3206#issuecomment-752510544 where I mentioned that `3.06`, `3.07` and `3.08` all had inconsistent versioning.

Now `3.09` also has inconsistent versioning:

1. `VERSION`: `3.08`
2. `setup.py`: `version='3.0.9'`
3. `src/LinearMath/btScalar.h`: `#define BT_BULLET_VERSION 307`
4. `src/LinearMath/btSerializer.h`:
```
buffer[9] = '3';
buffer[10] = '0';
buffer[11] = '8';
```

I'm not sure if there are more locations, but every single bullet release seems to miss at least one of those.

Maybe the release process could be documented so that those aren't forgotten, or the number of locations that hardcode version information could be reduced?

In `master` currently, there is still outdated information in above locations 1, 3, and 4 (same values, mix of `3.08` and `3.07`), while `setup.py` now shows `pybullet` as version `3.1.6`!

It seems pybullet gets a version bump regularly (pip updates?) and seems to be the main versioning for the whole of bullet nowadays?

Additionally, the `3.0.9` release on GitHub is shown as released on 26 Nov 2020: https://github.com/bulletphysics/bullet3/releases/tag/3.09
But it's tagged on a commit from 4 Mar 2021: https://github.com/bulletphysics/bullet3/tree/3.09

I guess it might have happened if you repurposed a pre-existing release to change its tag and name, instead of copy-pasting the release notes (which seem to be the same as `3.0.7` and `3.0.8` successively had). This makes it hard to know what changed between each release for downstream users.

**Comment by erwincoumans** [comment]
Yes, those releases are primarily pybullet versions. will update the c++ versions. I should have kept those versions separate, but some conda guys were confused about 2 release versions (c++ and pybullet).

**Reply by erwincoumans** [reply]
The versions are now 3.1.7 / 3.17
You'll need to check out the commits to see what changed.
